src/atomate2/vasp/schemas/defect.py

```diff
@@ -0,0 +1,377 @@
+            nbands=self.nbands,
+            data=np.array(self.data),
+        )
+
+
+class FiniteDiffDocument(BaseModel):
+    """Collection of computed WSWQDocuments using a single ref WAVECAR and a list of distorted WAVECARs."""
+
+    wswq_documents: List[WSWQDocument]
+    dir_name: str = Field(
+        None, description="Directory where the WSWQ calculations are performed"
+    )
+    ref_dir: str = Field(
+        None, description="Directory where the reference W(0) wavefunction comes from"
+    )
+    distorted_dirs: List[str] = Field(
+        None,
+        description="List of directories where the distorted W(Q) wavefunctions come from",
+    )
+
+    @classmethod
+    def from_directory(cls, directory: str | Path, **kwargs) -> FiniteDiffDocument:
+        """
+        Read the FintieDiff file.
+
+        Parameters
+        ----------
+        directory : str | Path
+            Path to the FintieDiff directory.
+        ref_dir : str
+            Directory where the reference W(0) wavefunction comes from.
+        distorted_dirs : List[str]
+            List of directories where the distorted W(Q) wavefunctions come from.
+
+        Returns
+        -------
+        FintieDiffDocument
+            FintieDiffDocument object.
+        """
+        wswq_dir = Path(directory)
+        files = list(Path(wswq_dir).glob("WSWQ.[0-9]*"))
+        ordered_files = sorted(files, key=lambda x: int(x.name.split(".")[1]))
+        wswq_documents = []
+        for f in ordered_files:
+            wswq_documents.append(WSWQDocument.from_file(f))
+
+        return cls(wswq_documents=wswq_documents, dir_name=str(wswq_dir), **kwargs)
+
+
+class CCDDocument(BaseModel):
+    """Configuration-coordiante definition of configuration-coordinate diagram."""
+
+    q1: int = Field(None, description="Charge state 1.")
+    q2: int = Field(None, description="Charge state 2.")
+    structure1: Structure = Field(
+        None, description="The structure of defect (supercell) in charge state (q2)."
+    )
+    structure2: Structure = Field(
+        None, description="The structure of defect (supercell) in charge state (q2)."
+    )
+
+    distortions1: List[float] = Field(
+        None,
+        description="The distortions of the defect (supercell) in charge state (q1).",
+    )
+    distortions2: List[float] = Field(
+        None,
+        description="The distortions of the defect (supercell) in charge state (q2).",
+    )
+
+    energies1: List[float] = Field(
+        None, description="The energies of the defect (supercell) in charge state (q1)."
+    )
+    energies2: List[float] = Field(
+        None, description="The energies of the defect (supercell) in charge state (q2)."
+    )
+
+    distorted_calcs_dirs: List[List[str]] = Field(
+        None,
+        description="Directories of distorted calculations, stored as seperate lists for each charge state",
+    )
+
+    @classmethod
+    def from_distorted_calcs(
```

Comment:
As mentioned above, it will be more efficient to specify the information you actually need. Which to me looks like just the structure, energy and directory. I'd also add the uuids for easier book keeping.